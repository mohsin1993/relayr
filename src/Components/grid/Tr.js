import React from 'react';
import PropTypes from 'prop-types';
import format from 'date-fns/format';

const Tr = ({
  name,
  unit,
  value,
  timestamp,
  active,
  updating,
  toggleActive,
}) => {
  return (
    <tr>
      <td>{name}</td>
      <td>{unit}</td>
      <td>{value}</td>
      <td>{format(new Date(timestamp), 'DD-MM-YYYY HH:mm')}</td>
      <td className='actions'>
        <input
          type='checkbox'
          checked={active}
          disabled={updating}
          onChange={() => toggleActive(name, !active)}
        />
        {updating && <span>{active ? 'deactivating' : 'activating'}...</span>}
      </td>
    </tr>
  );
};

Tr.propTypes = {
  name: PropTypes.string,
  unit: PropTypes.string,
  value: PropTypes.number,
  timestamp: PropTypes.number,
  active: PropTypes.bool,
  updating: PropTypes.bool,
  toggleActive: PropTypes.func.isRequired,
};

Tr.defaultProps = {
  name: '',
  unit: '',
  value: 0,
  timestamp: 0,
  active: false,
  updating: false,
};

export default Tr;
