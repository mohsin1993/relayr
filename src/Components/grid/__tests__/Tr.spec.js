import React from 'react';
import { mount } from 'enzyme';
import snapshot from 'check-snapshot';
import Tr from '../Tr';

describe('Tr', () => {
  let toggleActive = jest.fn();
  let props;
  beforeEach(() => {
    props = {
      name: 'acceleration_x',
      unit: 'm/s2',
      value: 25.993848858558,
      timestamp: 1558999341929,
      active: true,
      updating: false,
      toggleActive: toggleActive,
    };
  });

  it('It renders table row correctly', () => {
    snapshot(
      <table>
        <tbody>
          <Tr {...props} />
        </tbody>
      </table>
    );
  });

  it('It renders table row correctly', () => {
    snapshot(
      <table>
        <tbody>
          <Tr {...props} updating />
        </tbody>
      </table>
    );
  });

  it('onCheckbox change triggers toggleActive', () => {
    const wrapper = mount(
      <table>
        <tbody>
          <Tr {...props} />
        </tbody>
      </table>
    );
    wrapper
      .find('input[type="checkbox"]')
      .simulate('change', { target: { checked: !props.active } });
    expect(toggleActive).toHaveBeenCalledWith(props.name, !props.active);
  });
});
