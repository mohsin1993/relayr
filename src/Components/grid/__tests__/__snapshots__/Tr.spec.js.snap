// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tr It renders table row correctly 1`] = `
<table>
  <tbody>
    <tr>
      <td>
        acceleration_x
      </td>
      <td>
        m/s2
      </td>
      <td>
        25.993848858558
      </td>
      <td>
        28-05-2019 01:22
      </td>
      <td
        className="actions"
      >
        <input
          checked={true}
          disabled={false}
          onChange={[Function]}
          type="checkbox"
        />
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Tr It renders table row correctly 2`] = `
<table>
  <tbody>
    <tr>
      <td>
        acceleration_x
      </td>
      <td>
        m/s2
      </td>
      <td>
        25.993848858558
      </td>
      <td>
        28-05-2019 01:22
      </td>
      <td
        className="actions"
      >
        <input
          checked={true}
          disabled={true}
          onChange={[Function]}
          type="checkbox"
        />
        <span>
          deactivating
          ...
        </span>
      </td>
    </tr>
  </tbody>
</table>
`;
